source/agora/network/Manager.d

```diff
@@ -996,14 +996,16 @@ public class NetworkManager
         import std.typecons;
 
         // block count to check before the current block
-        const recent_block_count = 2;
+        const blocks_since_last_payment = ledger.getBlockHeight % ledger.params.PayoutPeriod;
 
         size_t[Height] signed_validators;
 
         try
         {
-            auto start_height = Height(max(1, ledger.getBlockHeight - recent_block_count));
-            BlockHeader[] headers = ledger.getBlocksFrom(start_height).map!(block => block.header).array;
+            auto start_height = Height(blocks_since_last_payment < ledger.getBlockHeight ? 1
```

Comment:
```suggestion
            auto start_height = Height(blocks_since_last_payment >= ledger.getBlockHeight ? 1
```
Isn't it the right formula?

Reply:
I think it is right but will add unit test to make sure.

Reply:
Fixed with unit test added.